pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -741,6 +751,39 @@ func (c *Controller) AppendInstanceHandler(f func(*model.ServiceInstance, model.
 		}
 		return nil
 	})
+
+	c.services.handler.Append(func(obj interface{}, event model.Event) error {
+		svc, ok := obj.(*v1.Service)
+		if !ok {
+			return nil
+		}
+
+		if svc.Namespace == meta_v1.NamespaceSystem {
+			return nil
+		}
+
+		svcConv := convertService(*svc, c.domainSuffix)
```

Comment:
cant you skip all this by checking apriori if this is an external name service?

Reply:
Done